VCCrypto/VCCrypto/Algo/HmacSha2AesCbc.swift

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import Foundation
+import CommonCrypto
+
+enum HmacSha2AesCbcError: Error {
+    case unsupportedMethod(name: String?)
+    case invalidAuthenticationCode
+}
+
+public struct HmacSha2AesCbc {
+
+    private let aes: Aes
+    private let hmac: HmacSha2
+    
+    public init(methodName: String) throws {
+        
+        let (hmacAlg, _) = try HmacSha2AesCbc.props(for: methodName)
+        self.hmac = try HmacSha2(algorithm: UInt32(hmacAlg))
+        self.aes = Aes()
+    }
+    
+    public func encrypt(plainText: Data, using aad: Data, iv: Data, with keys: (mac: EphemeralSecret, enc: EphemeralSecret)) throws -> (Data, Data) {
+        
+        let cipherText = try aes.encrypt(data: plainText, with: keys.enc, iv: iv)
+        let mac = try self.authenticate(aad: aad, iv: iv, cipherText: cipherText, with: keys.mac)
+        return (cipherText, mac)
+    }
+    
+    public func decrypt(_ input: (cipherText: Data, mac: Data), using aad: Data, iv: Data, with keys: (mac: EphemeralSecret, enc: EphemeralSecret)) throws -> Data {
+        
+        // Validate the message authentication code
+        let mac = try self.authenticate(aad: aad, iv: iv, cipherText: input.cipherText, with: keys.mac)
+        if input.mac == mac {
+            return try aes.decrypt(data: input.cipherText, with: keys.enc, iv: iv)
+        }
+        throw HmacSha2AesCbcError.invalidAuthenticationCode
+    }
+    
+    public static func props(for methodName: String) throws -> (Int, Int) {
```

Comment:
Because the type of the return value is not referenced anywhere other than in the function signature, I did not think it needed a name which wouldn't be used anywhere